crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1154,7 +1183,7 @@ impl<'db> Type<'db> {
                 left.is_equivalent_to(db, right)
             }
             (Type::Tuple(left), Type::Tuple(right)) => left.is_equivalent_to(db, right),
-            _ => self.is_fully_static(db) && other.is_fully_static(db) && self == other,
+            _ => self == other && self.is_fully_static(db) && other.is_fully_static(db),
```

Comment:
we have a [teeny tiny speedup on Codspeed](https://codspeed.io/astral-sh/ruff/branches/alex%2Fnested-tuples), which I suspect is just due to the change on this line